consensus/weight_test.go

```diff
@@ -0,0 +1,155 @@
+package consensus_test
+
+import (
+	"context"
+	"math/big"
+	"testing"
+
+	"github.com/ipfs/go-hamt-ipld"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/go-filecoin/address"
+	"github.com/filecoin-project/go-filecoin/consensus"
+	"github.com/filecoin-project/go-filecoin/state"
+	"github.com/filecoin-project/go-filecoin/types"
+)
+
+func TestNewWeight(t *testing.T) {
+	cst := hamt.NewCborStore()
+	ctx := context.Background()
+	fakeTree := &state.MockStateTree{}
+	// We only care about total power for the weight function
+	// Total is 16, so bitlen is 5
+	as := consensus.NewFakeActorStateStore(types.NewBytesAmount(1), types.NewBytesAmount(16), make(map[address.Address]address.Address))
+	tickets := []types.Ticket{consensus.MakeFakeTicketForTest()}
+	toWeigh := types.RequireNewTipSet(t, &types.Block{
+		ParentWeight: 0,
+		Tickets:      tickets,
+	})
+	sel := consensus.NewChainSelector(cst, as, types.CidFromString(t, "genesisCid"))
+
+	t.Run("basic happy path", func(t *testing.T) {
+		// 0 + 1[2*1 + 5] = 7
+		fixWeight, err := sel.NewWeight(ctx, toWeigh, fakeTree)
+		assert.NoError(t, err)
+		intWeight := requireFixedToInt(t, fixWeight)
+		assert.Equal(t, 7, intWeight)
+	})
+
+	t.Run("total power adjusts as expected", func(t *testing.T) {
+		asLowerX := consensus.NewFakeActorStateStore(types.NewBytesAmount(1), types.NewBytesAmount(15), make(map[address.Address]address.Address))
+		asSameX := consensus.NewFakeActorStateStore(types.NewBytesAmount(1), types.NewBytesAmount(31), make(map[address.Address]address.Address))
+		asHigherX := consensus.NewFakeActorStateStore(types.NewBytesAmount(1), types.NewBytesAmount(32), make(map[address.Address]address.Address))
+
+		// Weight is 1 lower than with total = 16 with total = 15
+		selLower := consensus.NewChainSelector(cst, asLowerX, types.CidFromString(t, "genesisCid"))
+		fixWeight, err := selLower.NewWeight(ctx, toWeigh, fakeTree)
+		assert.NoError(t, err)
+		intWeight := requireFixedToInt(t, fixWeight)
+		assert.Equal(t, 6, intWeight)
```

Comment:
Consider factoring out an `assertEqualInt(t, int, fixed)` to compress this noise.